### ISSUE TITLE
error:  abstract function type default value

### ISSUE THREAD

**Issue by hellobbq** [zephir-lang/zephir#1667]
```
# zephir
abstract class TestAbstract {
	abstract protected function testFunc(array text = [], string text2 = "", boolean flag = true);
}

# php:
class Test extends TestAbstract {
	protected function testFunc(array $text = [], string $text2 = "", bool $flag = true) {
		// do something...
	}
}

# catch error:
Declaration of Test::testFunc($text, $text2, $flag = true) must be compatible with TestAbstract::testFunc(?array $text, ?string $text2 = NULL, ?bool $flag = NULL)
```

This happens after the update:
form:
``
zephir: 0.10.7
zephir-parser: 1.1.1
``
to:
```
zephir: 0.11.0
zephir-parser: 1.1.3 or 1.1.1
```

centos6 php: 7.1

**Comment by hellobbq** [comment]
@danhunsaker i found this issues: https://github.com/phalcon/zephir/issues/457
any solution?

**Comment by danhunsaker** [comment]
The two aren't related. Sorry I haven't had a chance to look into this one more closely, yet, but it's related to how we're handling type hinting in `0.11.0`.

**Comment by Jurigag** [comment]
Then we need to revert it really because it's causing a lot of problems, i've seen the same issues on phalcon repository i think.

**Comment by sergeyklay** [comment]
@Jurigag Phalcon does not use 0.11.x branch. @hellobbq Could you downgrade to the [0.10.11](https://github.com/phalcon/zephir/releases/tag/0.10.11) and test again?

**Comment by sergeyklay** [comment]
Just fixed this bug for 0.11.x branch in the #1685
